lib/tools/apk-utils.js

```diff
@@ -0,0 +1,165 @@
+import { buildStartCmd, getActivityRelativeName } from '../helpers.js';
+import log from '../logger.js';
+import _ from 'lodash';
+import { sleep } from 'asyncbox';
+
+let apkUtilsMethods = {};
+
+apkUtilsMethods.isAppInstalled = async function (pkg) {
+  try {
+    let installed = false;
+    log.debug(`Getting install status for ${pkg}`);
+    let apiLevel = await this.getApiLevel();
+    let thirdparty = apiLevel >= 15 ? "-3" : "";
+    let stdout = await this.shell(['pm', 'list', 'packages', thirdparty, pkg]);
+    let apkInstalledRgx = new RegExp('^package:' + pkg.replace(/(\.)/g, "\\$1") +
+                                     '$', 'm');
+    installed = apkInstalledRgx.test(stdout);
+    log.debug("App is" + (!installed ? " not" : "") + " installed");
+    return installed;
+  } catch (e) {
+    log.errorAndThrow(`Error finding if app is installed. Original error: ${e.message}`);
+  }
+};
+
+apkUtilsMethods.startApp = async function (startAppOptions = {}) {
+  try {
+    if (!startAppOptions.activity || !startAppOptions.pkg) {
+      log.errorAndThrow("activity and pkg is required for launching application");
+    }
+    // initializing defaults
+    _.defaults(startAppOptions, {
```

Comment:
Better clone the options before using defaults. It is modifying startAppOptions. 
(https://lodash.com/docs#defaults)